core/templates/dev/head/services/ExplorationFeaturesService.ts

```diff
@@ -17,44 +17,52 @@
  *               the exploration editor.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-oppia.factory('ExplorationFeaturesService', [function() {
-  var settings = {
-    areParametersEnabled: false,
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationFeaturesService {
+  static settings = {
     isImprovementsTabEnabled: false,
     isPlaythroughRecordingEnabled: false,
-  };
+    areParametersEnabled: false
+  }
 
-  return {
-    init: function(explorationData, featuresData) {
-      settings.isImprovementsTabEnabled =
-        featuresData.is_improvements_tab_enabled;
-      settings.isPlaythroughRecordingEnabled =
-        featuresData.is_exploration_whitelisted;
-      if (explorationData.param_changes &&
-          explorationData.param_changes.length > 0) {
-        this.enableParameters();
-      } else {
-        for (var state in explorationData.states) {
-          if (explorationData.states[state].param_changes.length > 0) {
-            this.enableParameters();
-            break;
-          }
+  init(explorationData, featuresData) {
+    ExplorationFeaturesService.settings.isImprovementsTabEnabled =
```

Comment:
Is there a way to say cls. or this. instead of ExplorationFeatureService? I am not familiar with angular 8, so I don't know. Just asking.

Reply:
I tried using `this.` but it wasn't doing the right thing. This was the only way I got this worked.